tests/treemodel/delta/movedelta.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * @license Copyright (c) 2003-2015, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/* bender-tags: treemodel, delta */
+
+'use strict';
+
+const modules = bender.amd.require(
+	'treemodel/document',
+	'treemodel/position',
+	'treemodel/range',
+	'treemodel/element'
+);
+
+describe( 'Batch', () => {
+	let Document, Position, Range, Element;
+
+	let doc, root, div, p, batch, chain;
+
+	before( () => {
+		Document = modules[ 'treemodel/document' ];
+		Position = modules[ 'treemodel/position' ];
+		Range = modules[ 'treemodel/range' ];
+		Element = modules[ 'treemodel/element' ];
+	} );
+
+	beforeEach( () => {
+		doc = new Document();
+		root = doc.createRoot( 'root' );
+
+		div = new Element( 'div', [], 'foobar' );
+		p = new Element( 'p', [], 'abcxyz' );
+
+		root.insertChildren( 0, [ div, p ] );
+
+		batch = doc.batch();
+	} );
+
+	function getNodesAndText( element ) {
+		let range = Range.createFromElement( element );
+		let txt = '';
+
+		for ( let step of range ) {
+			let node = step.node;
+
+			if ( node.character ) {
+				txt += node.character.toLowerCase();
+			} else if ( node.name ) {
+				txt += node.name.toUpperCase();
+			}
+		}
+
+		return txt;
+	}
+
+	describe( 'moveNode', () => {
+		beforeEach( () => {
+			chain = batch.moveNode( div, new Position( root, [ 2 ] ) );
+		} );
+
+		it( 'should move specified node', () => {
+			expect( root.getChildCount() ).to.equal( 2 );
+			expect( getNodesAndText( root.getChild( 0 ) ) ).to.equal( 'abcxyz' );
+			expect( getNodesAndText( root.getChild( 1 ) ) ).to.equal( 'foobar' );
+		} );
+
+		it( 'should be chainable', () => {
+			expect( chain ).to.equal( batch );
+		} );
+	} );
+
+	describe( 'move', () => {
+		beforeEach( () => {
+			div.insertChildren( 4, [ new Element( 'p', [], 'gggg' ) ] );
+			div.insertChildren( 2, [ new Element( 'p', [], 'hhhh' ) ] );
```

Comment:
It was a good order after all, if you think that `div` content was equal to `foobar`. If I want to insert after first "o" (offset 2) and after "b"  (offset 4) I'd have to either insert at 2 then at **5** or do it backwards and insert at original offsets. I like to think about it this way so I don't have to re-calculate offsets, but for a moment I though it was incorrect (hence the comment before)